src/test/java/org/springframework/data/elasticsearch/core/SearchTemplateIntegrationTests.java

```diff
@@ -63,83 +65,264 @@ public abstract class SearchTemplateIntegrationTests {
+		assertThat(multiSearchHits.get(1).getTotalHits()).isEqualTo(2);
+
+		assertThat(multiSearchHits.get(0).getSearchHits())
+				.extracting(SearchHit::getContent)
+				.extracting(Person::lastName)
+				.contains("Smith", "Myers");
+		assertThat(multiSearchHits.get(1).getSearchHits())
+				.extracting(SearchHit::getContent)
+				.extracting(Person::firstName)
+				.contains("John", "Willy");
+
+		assertThat(operations.deleteScript(SCRIPT_SEARCH_FIRSTNAME.id())).isTrue();
+		assertThat(operations.deleteScript(SCRIPT_SEARCH_LASTNAME.id())).isTrue();
+	}
+
+	@Test // #2704
+	@DisplayName("should search with template multisearch with multiple classes")
+	void shouldSearchWithTemplateMultiSearchWithMultipleClasses() {
+		assertThat(operations.putScript(SCRIPT_SEARCH_FIRSTNAME)).isTrue();
+		assertThat(operations.putScript(SCRIPT_SEARCH_LASTNAME)).isTrue();
+
+		var q1 = SearchTemplateQuery.builder() //
+				.withId(SCRIPT_SEARCH_FIRSTNAME.id()) //
+				.withParams(Map.of("firstName", "John")) //
+				.build();
+		var q2 = SearchTemplateQuery.builder() //
+				.withId(SCRIPT_SEARCH_FIRSTNAME.id()) //
+				.withParams(Map.of("firstName", "Joey")) //
+				.build();
+
+		// search with multiple classes
+		var multiSearchHits = operations.multiSearch(List.of(q1, q2), List.of(Person.class, Student.class));
+
+		assertThat(multiSearchHits.size()).isEqualTo(2);
+		assertThat(multiSearchHits.get(0).getTotalHits()).isEqualTo(2);
+		assertThat(multiSearchHits.get(1).getTotalHits()).isEqualTo(1);
+
+		assertThat(multiSearchHits.get(0).getSearchHits())
+				// type casting is needed here
+				.extracting(hits -> (Person) hits.getContent())
+				.extracting(Person::lastName)
+				.contains("Smith", "Myers");
+		assertThat(multiSearchHits.get(1).getSearchHits())
+				// type casting is needed here
+				.extracting(hits -> (Student) hits.getContent())
+				.extracting(Student::lastName)
+				.containsExactly("Dunlop");
+
+		assertThat(operations.deleteScript(SCRIPT_SEARCH_FIRSTNAME.id())).isTrue();
+		assertThat(operations.deleteScript(SCRIPT_SEARCH_LASTNAME.id())).isTrue();
+	}
+
+	@Test // #2704
+	@DisplayName("should search with template multisearch with multiple index coordinates")
+	void shouldSearchWithTemplateMultiSearchWithMultipleIndexCoordinates() {
+		assertThat(operations.putScript(SCRIPT_SEARCH_FIRSTNAME)).isTrue();
+		assertThat(operations.putScript(SCRIPT_SEARCH_LASTNAME)).isTrue();
+
+		var q1 = SearchTemplateQuery.builder() //
+				.withId(SCRIPT_SEARCH_FIRSTNAME.id()) //
+				.withParams(Map.of("firstName", "John")) //
+				.build();
+		var q2 = SearchTemplateQuery.builder() //
+				.withId(SCRIPT_SEARCH_LASTNAME.id()) //
+				.withParams(Map.of("lastName", "Dunlop")) //
+				.build();
+
+		// search with multiple index coordinates
+		var multiSearchHits = operations.multiSearch(List.of(q1, q2), List.of(Person.class, Student.class),
+				List.of(IndexCoordinates.of(indexNameProvider.indexName()), IndexCoordinates.of("another-index")));
```

Comment:
```suggestion
		var multiSearchHits = operations.multiSearch(
				List.of(q1, q2),
				List.of(Person.class, Student.class),
				List.of(IndexCoordinates.of(indexNameProvider.indexName() + "-person"),
						IndexCoordinates.of(indexNameProvider.indexName() + "-student")));
```

Reply:
Is there any code guideline for code formatting? I fonud [this](https://github.com/spring-projects/spring-data-build/blob/main/CONTRIBUTING.adoc) in the contributing file, but seems that this is not the format finally accepted.

Plus: Any ideas why this project uses tabs rather than 4 spaces? That's really rare as far as i'm concerned.